doc: nrf: app_dev: Update files to show dts file output (#29149)
nrfconnect/sdk-nrf: doc/nrf/app_dev/programming.rst

|application_sample_definition|

> To program the :ref:`output build files <app_build_output_files>` to your device, which in most of the cases will be :file:`zephyr.hex` or :file:`merged_<board_target>.hex`, use the steps for the development environment.

Reviewer: ```suggestion
To program the :ref:`output build files <app_build_output_files>` to your device, use the steps for your development environment.

In most cases, the relevant file is :file:`zephyr.hex` or :file:`merged_<board_target>.hex`.
```

Author: fixed
